### PR TITLE
React and Vite scaffolding

### PR DESCRIPTION
**Pull request by TylerMatteo**
This PR is meant to be the minimum project scaffolding to starting setting up this repo. Although the plan is to eventually use Storybook, Storybook itself does not handle concerns like transpiling and bundling. To set up that basic infrastructure, I used [Vite](https://vitejs.dev/guide/#scaffolding-your-first-vite-project), which is [officially supported](https://storybook.js.org/docs/react/builders/vite#page-top) by Storybook. This PR includes the result of running `npm create vite@latest . -- --template react-ts`, with the following additional changes:

* Removing the boilerplate README that Vite adds and replacing it with our own.
* Adding a `.nvmrc` file pointing to the current node 18 LTS version (18.17.1).

Other config and dot files seen in this PR, such as `.gitignore` and `tsconfig.json`, have been intentionally left as exactly what the Vite generator outputs.

Proper code quality tooling will be included in a subsequent PR. This PR installs Eslint only because it comes with Vite's generator.

Resolves AD14802.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,3 @@
-# ae-design-system
-Our design system implementation. This repo outputs a Storybook site as well as an NPM package of reusable components and tokens.
+# Design System
+
+This repo contains our primary design system implementation. For now, it contains a basic React template [generated using Vite](https://vitejs.dev/guide/#scaffolding-your-first-vite-project) with the `react-ts` template.
```
Comment (TangoYankee) [inline README.md:3]:
```suggestion
This repo contains the primary design system for DCP applications. For now, it contains a basic React template [generated using Vite](https://vitejs.dev/guide/#scaffolding-your-first-vite-project) with the `react-ts` template.
```

Comment (TylerMatteo) [reply README.md:3]: I can appreciate that my wording here is a little opaque but I don't think we should say that this repository _is_ the design system. The standards, components, guidelines, and best practices that make up a design system exist outside of this repo.

**Comment by TangoYankee** [comment]
Do we want to change the default strategy to rebasing?

**Review by TangoYankee — APPROVED** [review]

**Comment by TylerMatteo** [comment]
> 

There doesn't seem to be a way to configure the default strategy :/ but I did disable merge commits.

**Pull request merged**